### ISSUE TITLE
Feature request: Detect if camera is disconnected or stopped

### ISSUE THREAD

**Issue by fb39ca4** [inspirit/PS3EYEDriver#31]
Could be used in a situation where the user might hotplug cameras.

**Comment by mkalten** [comment]
I'd also be interested in this feature ...
currently the library just stays locked in getFrame() when a camera is disconnected.
would be best to return silently and set is_running to false!

**Comment by inspirit** [comment]
this is handled on usb driver level. as far as i remember there was no way to handle it inside PS3EYEDriver

**Comment by mkalten** [comment]
well, I guess it should be somehow possible to work around this.
I had a look at the current code, and currently it stays locked at empty_condition.wait after a camera is disconnected. on the other hand, transfer_canceled() is also called, when the camera is unplugged, so maybe this special case could be handled from there?

**Comment by inspirit** [comment]
can you try to run the test with additional condition for the lock? unfortunately i dont have camera at the moment and cant verify

**Comment by mkalten** [comment]
I hacked a quick and dirty workaround for detecting a disconnected camera.
In general this works quite fine for my needs, although I had to tweak the shared_ptr delete in order to avoid a crash when cleaning up everything ...
https://github.com/mkalten/PS3EYEDriver/commit/660106b7f9fb74cb767bd11a6fdf86e9c9f66872

**Comment by inspirit** [comment]
uhm, thats a bit too harsh to remove deleter from shared_ptr and i dont see where you will be deleting it later. i believe there can be cleaner solution...

**Comment by mkalten** [comment]
yes I know, that's the only part which I didn't figure out yet, that's why I asked.
maybe you can find a better solution for a clean exit, 
but the disconnect detection part works quite reliably apart from that ...
